### PR TITLE
Feat-9: Add interactive renderers

### PR DESCRIPTION
**Pull request by tussharrane**

### CODE REVIEW

**Comment by kkyusufk** [comment]
Okay, here are some thoughts:
- the first 3 vidoes and next 3 mcqs are all part of the same lessons. so I do no expect there to be a drastic change in the UI when switching from video to mcq. I do not see the progress bar, the forward and backward button on mcq screens. 
- I dont expect a seperate screen for showing the correct answer, just a dimmer on screen with a tick mark can be enough.
- I see that when you are developing these new features you do not ask or question anything, you are assuming a lot of things and just doing stuff on your own. Please be more curious and ask more questions in future.

**Comment by tussharrane** [comment]
@kkyusufk 
Okay ,
So, should I keep this MCQRenderer component and add similar styles as of the video screens. ?
Or, should I render the MCQRenderer instead of the video element inside of the Screen.

**Comment by kkyusufk** [comment]
> Or, should I render the MCQRenderer instead of the video element inside of the Screen.

with the mcq screens, the lesson now has 6 screens. Depending on that the progress bar should update. So it would fill up after 6 screens. Also try to mix and match, first screen video, next 2 screens mcqs then video. Make one data.json which might be easier.

**Comment by tussharrane** [comment]
@kkyusufk 
I have added the mcqRenderer  and data.json as we discussed.
I noticed a bug, If we go to the previous screens and re-select the mcq answers, the score increases. 
For this, should I create a global state variable which tracks which mcq question has been marked correct, and so the score wont go more than the number of question.

**Comment by kkyusufk** [comment]
> For this, should I create a global state variable which tracks which mcq question has been marked correct, and so the score wont go more than the number of question.

Yup, something like that
But lets track it in a seperate piece of work, I would just mark the screen as completed once its finished.

image
this score thing at the top, it should be removed

**Comment by kkyusufk** [comment]
image
pause button is disabled in mc renderers

**Comment by kkyusufk** [comment]
forward button is disabled on last screen but it is still hover-able. It should not be hover-able

**Comment by kkyusufk** [comment]
Just a heads up,
Whenever you add new features, do a manual play through and make sure new features did not break old working features

File: src/Components/McqRenderer/McqRenderer.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import React, { useState } from 'react';
+import style from './McqRenderer.module.css';
+import Dimmer from '../Dimmer/Dimmer';
+
+const McqRenderer = ({ data, onEnd, showContent }) => {
+	const [showSubmitBtn, setShowSubmitBtn] = useState(false);
```
Comment (kkyusufk) [inline src/Components/McqRenderer/McqRenderer.js:6]: `showSubmitButton`

File: src/Components/MainContainer/MainContainer.js [hunk 2]
```diff
@@ -15,10 +15,10 @@ const MainContainer = () => {
 	const [isLessonClosed, setIsLessonClosed] = useState(false);
 
 	useEffect(() => {
-		dispatchFn(actions.setTotalScreens(videos.length));
+		dispatchFn(actions.setTotalScreens(data.length));
 	}, []);
 
-	const onCurrVideoEnded = () => {
+	const onCurrScreenEnded = () => {
```
Comment (kkyusufk) [inline src/Components/MainContainer/MainContainer.js:21]: onCurrentScreenEnded

File: src/store/actionCreators.js [hunk 3]
```diff
@@ -13,10 +13,12 @@ const setTotalScreens = (screens) => ({
 	type: actions.TOTAL_SCREENS,
 	payload: screens,
 });
+const incrementScore = () => ({ type: actions.ADD_SCORE });
```
Comment (kkyusufk) [inline src/store/actionCreators.js:16]: out of scope of this change, but you can destructure all these actions above and use them.
ex: `import { ADD_SCORE } from './actions'`

**Review by kkyusufk — CHANGES_REQUESTED** [review]

**Comment by tussharrane** [comment]
> pause button is disabled in mc renderers
@kkyusufk 
In mcq, play/pause button has no function as there is no video to play or pause. 
Should I keep it disabled and if not, what should the pause/play button do?

**Comment by kkyusufk** [comment]
> > pause button is disabled in mc renderers
> 
> @kkyusufk In mcq, play/pause button has no function as there is no video to play or pause. Should I keep it disabled and if not, what should the pause/play button do?

play-pause should disable the UI, thats what the functionality should be

**Comment by tussharrane** [comment]
> play-pause should disable the UI, thats what the functionality should be

@kkyusufk 
should the mcq Options be disabled as well when paused? or just the next and previous buttons..

**Comment by tussharrane** [comment]
@kkyusufk 
When paused the ui buttons are disabled also a Dimmer to show the screen is Paused just like in Video Screens.

**Review by kkyusufk — APPROVED** [review]

**Pull request merged**